File: Models/Student.cs
using Almond.LearningCentre.Models.Enum;

namespace Almond.LearningCentre.Models
{
    public partial class Student: AbstractEntity
    {
        public Student()
        {
            Assessments = new HashSet<Assessment>();
            StudentCourses = new HashSet<StudentCourse>();
        }


        public string ImageUrl { get; set; } = null!;
        public string FirstName { get; set; } = null!;
        public string Surname { get; set; } = null!;
        public int Grade { get; set; }
        public DateTime DateOfBirth { get; set; }
        public string GuardianName { get; set; } = null!;
        public string GuardianAddress { get; set; } = null!;
        public string GuardianPhoneNumber { get; set; } = null!;
        public string GuardianEmail { get; set; } = null!;
        public string Allergies { get; set; } = null!;
        public virtual ICollection<Assessment> Assessments { get; set; }
        public virtual ICollection<StudentCourse> StudentCourses { get; set; }
    }
}

File: wwwroot/js/Student.js
﻿var dataTable;

$(document).ready(function () {
    loadDataTable();
});

function loadDataTable() {
    dataTable = $('#tblData').DataTable({
        "ajax": {
            "url": "/Student/GetAll"
        },
        "columns": [
            { "data": "imageUrl", "width": "15%" },
            { "data": "firstName", "width": "15%" },
            { "data": "surname", "width": "15%" },
            { "data": "grade", "width": "15%" },
            { "data": "dateOfBirth", "width": "15%" },
            { "data": "guardianName", "width": "15%" },
            { "data": "guardianPhoneNumber", "width": "15%" },
            { "data": "guardianEmail", "width": "15%" },
            { "data": "allergies", "width": "15%" },
            { "data": "isDeleted", "width": "15%" },
            {
                "data": "id",
                "render": function (data) {
                    return `
                        <div class="w-75 btn-group" role="group">
                        <a href="/Student/Upsert?id=${data}"
                        class="btn btn-primary mx-2"> <i class="bi bi-pencil-square"></i> Edit</a>
                        <a onClick=Delete('/Student/Delete/${data}')
                        class="btn btn-danger mx-2"> <i class="bi bi-trash-fill"></i> Delete</a>
					</div>
                        `
                },
                "width": "15%"
            }
        ]
    });
}

function Delete(url) {
    Swal.fire({
        title: 'Are you sure?',
        text: "You won't be able to revert this!",
        icon: 'warning',
        showCancelButton: true,
        confirmButtonColor: '#3085d6',
        cancelButtonColor: '#d33', 
        confirmButtonText: 'Yes, delete it!'
    }).then((result) => {
        if (result.isConfirmed) {
            $.ajax({
                url: url,
                type: 'DELETE',
                success: function (data) {
                    if (data.success) {
                        dataTable.ajax.reload();
                        toastr.success(data.message);
                    }
                    else {
                        toastr.error(data.message);
                    }
                }
            })
        }
    })
}